### PR TITLE
datastore: Make it possible to replace events

### LINKED ISSUE #35: Stopwatch not working

### PR DESCRIPTION
**Pull request by johan-bjareholt**
- Sets the default duration to 0 if not specified
- Makes it possible to replace events
- insert_events now returns the events with an assigned id

### CODE REVIEW

File: tests/datastore.rs [hunk 1]
```diff
@@ -183,6 +183,33 @@ mod datastore_tests {
...
+            duration: Duration::seconds(0),
+            data: json_map!{"key": json!("value")}
+        };
+        ds.insert_events(&bucket.id, &[e1.clone()]).unwrap();
+
+        let mut fetched_events = ds.get_events(&bucket.id, None, None, None).unwrap();
+        assert_eq!(fetched_events.len(), 1);
+        let e2 = fetched_events.pop().unwrap();
+
+        ds.insert_events(&bucket.id, &[e2.clone()]).unwrap();
```
Comment (ErikBjare) [inline tests/datastore.rs]: You're not really modifying the event though?

Maybe try changing the duration and the data field?

**Comment by johan-bjareholt** [comment]
I cannot apparently deny my own pull request, but it actually seems like it's possible to get the lastrowid before a transaction is commited.

If it works I will rewrite this change instead to always return a list of all events with a id on all of them (looks possible, but not sure until I have tested).

**Comment by johan-bjareholt** [comment]
A couple of clones that needs to be fixed and some more testing, it is verified that it works though!

**Comment by johan-bjareholt** [comment]
Fixed, waiting for travis to finish then I'm probably merging

**Pull request merged**